### PR TITLE
Upgrades and upgrade tests take versions of the form release/stable instead of stable_release

### LINKED ISSUE #15506: Upgrade scripts and tests should take arbitrary published versions
Right now the [cluster_upgrade e2e](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/cluster_upgrade.go) takes:

```
// realVersion turns a version constant s--one accepted by cluster/gce/upgrade.sh--
// into a deployable version string. If the s is not known to be a version
// constant, it will assume it is already a valid version, and return s directly.
//
// NOTE: KEEP THIS LIST UP-TO-DATE WITH THE CODE BELOW.
// The version strings supported are:
// - "latest_stable"   (returns a string like "0.18.2")
// - "latest_release"  (returns a string like "0.19.1")
// - "latest_ci"       (returns a string like "0.19.1-669-gabac8c8")
```

And the [upgrade.sh](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/upgrade.sh) script takes similar params (via [set_binary_version](https://github.com/kubernetes/kubernetes/blob/master/cluster/common.sh#L234):

```
${0} [-M|-N|-P] -l | <release or continuous integration version> | [latest_stable|latest_release|latest_ci]
```

Furthermore, there is the script [hack/get-build.sh](https://github.com/kubernetes/kubernetes/blob/master/hack/get-build.sh), as documented in [Getting Kubernetes Builds](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/getting-builds.md).

These all need to take an arbitrary build string:

- `ci/latest`
- `ci/latest-1`
- `ci/latest-1.2`
- `release/latest`
- `release/latest-1`
- `release/latest-1.2`
- `release/stable`
- `release/stable-1`
- `release/stable-1.2`

So that we can test upgrades properly, and generally for functioning upgrades.

The logic should also be consolidated if possible.

### PR DESCRIPTION
**Pull request by ikehz**
Fixes #15506.  See the issue for more info.

### CODE REVIEW

**Comment by ihmccreery** [comment]
@mikedanese Would you mind reviewing this?  Sorry it's so big.

File: cluster/common.sh [hunk 1]
```diff
@@ -278,6 +290,10 @@ function tars_from_version() {
     echo "Failure trying to curl Salt tar .sha1"
   done
 
+  if ! curl -Ss --range 0-1 "${KUBE_TAR_URL}" >&/dev/null; then
```
Comment (mikedanese) [inline cluster/common.sh]: note: existence checks on gcs should use --head. This pattern was not introduced in this pr so we don't have to fix it here.

https://cloud.google.com/storage/docs/xml-api/head-object?hl=en

File: hack/get-build.sh [hunk 2]
```diff
@@ -18,53 +18,59 @@ set -o errexit
...
 declare -r KUBE_TAR_NAME="kubernetes.tar.gz"
 
 usage() {
-	echo "usage:
-  $0 [stable|release|latest|latest-green]
-
-        stable:        latest stable version
-        release:       latest release candidate
-        latest:        latest ci build
-        latest-green:  latest ci build to pass gce e2e"
```
Comment (mikedanese) [inline hack/get-build.sh]: this breaks compatibility, not a big deal?

**Comment by mikedanese** [comment]
Great cleanup, thanks! Please squash fixup commits then LGTM

File: hack/get-build.sh (hunk 2 above)
Comment (ihmccreery) [inline hack/get-build.sh]: If people are scripting off of this, their scripts will definitely break.  I thought that was a lesser evil than keeping these aliases around, but I'm definitely willing to be overruled.

File: cluster/common.sh (hunk 1 above)
Comment (ihmccreery) [inline cluster/common.sh]: Ack.

**Comment by ihmccreery** [comment]
Squashed; @mikedanese PTAL.

File: hack/get-build.sh (hunk 2 above)
Comment (mikedanese) [inline hack/get-build.sh]: I'm ok with it as well

**Comment by mikedanese** [comment]
LGTM

**Pull request merged**